xnmt/batcher.py

```diff
@@ -139,8 +141,12 @@ def cmult_by_timestep_expr(self, expr, timestep, inverse=False):
       mask_exp = dy.inputTensor(self.np_arr[:,timestep:timestep+1].transpose(), batched=True)
     return dy.cmult(expr, mask_exp)
 
-  def get_active_one_mask(self):
-    return 1 - self.np_arr
+  @lru_cache(maxsize=1)
```

Comment:
If so, then that's fine.

Reply:
I also think the mask was intended to be immutable, and haven't seen any code that modifies it. It might be good to put a short "immutable" comment in the doc string?

Reply:
Adding that